### PR TITLE
Create accounts and deploy contracts

### PR DESCRIPTION
**Pull request by bowenwang1996**
Enable the following:

- Create accounts

- Swap keys to transfer accounts

- Deploy contracts to account

### CODE REVIEW

File: core/primitives/src/types.rs [hunk 1]
```diff
@@ -119,13 +119,35 @@ pub struct FunctionCallTransaction {
     pub args: Vec<u8>,
 }
 
+#[derive(Hash, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct CreateAccountTransaction {
+    pub nonce: u64,
+    pub sender: AccountId,
+    // receiver is the account id to be created
```
Comment (azban) [inline core/primitives/src/types.rs]: these structs exist so we don't have to have comments like this. `new_account_id` works

File: core/primitives/src/types.rs [hunk 2]
```diff
@@ -119,13 +119,35 @@ pub struct FunctionCallTransaction {
...
+    pub public_key: Vec<u8>,
+}
+
+#[derive(Hash, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct SwapKeyTransaction {
+    pub nonce: u64,
+    pub sender: AccountId,
+    // current key to the account.
+    // sender must sign the transaction with this key
+    pub cur_key: Vec<u8>,
```
Comment (azban) [inline core/primitives/src/types.rs]: isn't this redundant? we will be checking that the sig matches what is on the account already

File: node/rpc/src/types.rs [hunk 3]
```diff
@@ -23,10 +23,29 @@ pub struct StakeRequest {
...
     #[serde(with = "bs58_format")]
-    pub owner_account_id: AccountId,
-    #[serde(with = "bs58_format")]
     pub contract_account_id: AccountId,
     pub wasm_byte_array: Vec<u8>,
+    pub public_key: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateAccountRequest {
```
Comment (azban) [inline node/rpc/src/types.rs]: again, i would be more verbose here rather than just reusing the transaction struct

File: scripts/rpc.py [hunk 4]
```diff
@@ -341,6 +384,36 @@ def deploy(self):
...
+        client = self._get_rpc_client(args)
+        return client.create_account(
+            args.sender,
+            args.receiver,
+            args.amount,
+            args.public_key
+        )
+
+    def swap_key(self):
+        """Create an account"""
```
Comment (azban) [inline scripts/rpc.py]: bad copypasta

File: core/primitives/src/types.rs [hunk 5]
```diff
@@ -135,15 +157,19 @@ impl TransactionBody {
...
     }
 
     pub fn get_sender(&self) -> AccountId {
         match self {
             TransactionBody::Stake(t) => t.staker,
             TransactionBody::SendMoney(t) => t.sender,
-            TransactionBody::DeployContract(t) => t.owner,
+            TransactionBody::DeployContract(t) => t.contract_id,
             TransactionBody::FunctionCall(t) => t.originator,
+            TransactionBody::CreateAccount(t) => t.sender,
```
Comment (azban) [inline core/primitives/src/types.rs]: i'm in favor of originator if you are not actually sending something like in `SendMoney`

Comment (bowenwang1996) [reply core/primitives/src/types.rs]: It does send money though

**Pull request merged**